### PR TITLE
Better logging + regal outfit

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
Sprite by Ayshe aka gid_git

New Crimson cross outfit 
***the first crimson cross outfit***
Regal greatecoat
This outfit makes you almost completely restant to robotic attacks, be it robots or FPBs!

**ADMIN*** Logs now properly lable what attacked someone even if its an NPC

### CODE REVIEW

**Comment by cdb-is-not-good** [comment]
what's the better logging? It's not described in your changelog.

File: code/modules/mob/living/carbon/human/human_attackhand.dm [hunk 1]
```diff
@@ -357,6 +357,11 @@
 
 	user.do_attack_animation(src)
 
+	//reduced damage
+	var/RD = uniquic_armor_check(null, user, damage)
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/human/human_attackhand.dm:361]:
```suggestion
	var/RD = unique_armor_check(null, user, damage)
```

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 2]
```diff
@@ -4,7 +4,7 @@ Contains most of the procs that are called when a mob is attacked by something
 bullet_act
 ex_act
 meteor_act
-
+uniquic_armor_act
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/human/human_defense.dm:7]:
```suggestion
unique_armor_act
```

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 3]
```diff
@@ -13,6 +13,8 @@ meteor_act
 	if(!has_organ(def_zone))
 		return PROJECTILE_FORCE_MISS //if they don't have the organ in question then the projectile just passes by.
 
+	uniquic_armor_check(P, null, null)
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/human/human_defense.dm:16]:
```suggestion
	unique_armor_check(P, null, null)
```

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 4]
```diff
@@ -233,7 +235,11 @@ meteor_act
 	if(!affecting)
 		return FALSE//should be prevented by attacked_with_item() but for sanity.
 
-	visible_message("<span class='danger'>[src] has been [I.attack_verb.len? pick(I.attack_verb) : "attacked"] in the [affecting.name] with [I.name] by [user]!</span>")
+	visible_message("<span class='danger'>[src] has been [I.attack_verb.len? pick(I.attack_verb) : "attacked"] in the  ffecting.name] with [I.name] by [user]!</span>")
+
+	var/EF = uniquic_armor_check(I, user, effective_force)
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/human/human_defense.dm:240]:
```suggestion
	var/EF = unique_armor_check(I, user, effective_force)
```

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 5]
```diff
@@ -489,3 +495,92 @@ meteor_act
 		perm += perm_by_part[part]
 
 	return perm
+
+
+//soj edit
+//This atm only has 1 armor in it thus its coding is trash and snowflake
+/mob/living/carbon/human/proc/uniquic_armor_check(atom/A, mob/user, EF)
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/human/human_defense.dm:502]:
```suggestion
/mob/living/carbon/human/proc/unique_armor_check(atom/A, mob/user, EF)
```

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 6]
```diff
@@ -489,3 +495,92 @@ meteor_act
...
+	//message_admins("uniquic_armor_check([user.name]) EF [EF]")
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/human/human_defense.dm:503]:
```suggestion
	//message_admins("unique_armor_check([user.name]) EF [EF]")
```

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 7]
```diff
@@ -489,3 +495,92 @@ meteor_act
...
+					//		message_admins("bluecross_regaloutfit Proj Pass - [Proj.original_firer] !!!!!!")
+							en_passant = TRUE
+							break
+					if(ishuman(Proj.original_firer))
+						var/mob/living/carbon/human/H = Proj.original_firer
+						if(H.species.reagent_tag == IS_SYNTHETIC)
+					//		message_admins("bluecross_regaloutfit Proj Pass")
+							en_passant = TRUE
+					if(en_passant)
+					//	message_admins("uniquic_armor_check en_passant ranged")
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/human/human_defense.dm:549]:
```suggestion
					//	message_admins("unique_armor_check en_passant ranged")
```

File: code/modules/projectiles/guns/oddity_items.dm [hunk 8]
```diff
@@ -541,4 +541,21 @@
 //Armor
 
 
+// Regal Outfit (Crimson Clothing)
+/obj/item/clothing/suit/crimsoncross_regaloutfit
+	name = "\"Regal\" greatecoat"
+	desc = "An anomalous weapon created by rivals of the unknown person(or group?) of the bluecross, their work marked by a crimson cross, these weapons are known to vanish and reappear when left alone. \
```
Comment (cdb-is-not-good) [inline code/modules/projectiles/guns/oddity_items.dm:547]:
```suggestion
	desc = "An anomalous weapon created by rivals of the unknown person(or group?) of the bluecross, their work marked by a crimson cross, these items are known to vanish and reappear when left alone. \
```

File: code/modules/projectiles/guns/oddity_items.dm [hunk 9]
```diff
@@ -541,4 +541,21 @@
...
+	A highly decorated white cloak with golden epaulets over a blue vest and sleek white pants.  It is absolutely stained in blood forevermore. An air of nobility ensures that any upraising of a robotic foe will be reduced."
```
Comment (cdb-is-not-good) [inline code/modules/projectiles/guns/oddity_items.dm:548]:
```suggestion
	A highly decorated white cloak with golden epaulets over a blue vest and sleek white pants.  It is forevermore stained with blood. An air of nobility ensures that no robotic foe may ever rise against its wearer."
```

**Review by cdb-is-not-good — COMMENTED** [review]
some spellchecks and a few grammar/wording suggestions. other than that, looks fine.

File: code/modules/projectiles/guns/oddity_items.dm [hunk 10]
```diff
@@ -541,4 +540,22 @@
 //Armor
 
 
+// Regal Outfit (Crimson Clothing)
+/obj/item/clothing/suit/crimsoncross_regaloutfit
+	name = "\"Regal\" greatecoat"
```
Comment (Trilbyspaceclone) [inline code/modules/projectiles/guns/oddity_items.dm:545]:
```suggestion
	name = "\"Regal\" greatcoat"
```

**Comment by ChefDoggo** [comment]
still heavily against this as its insanely unbalanced with basically zero downsides whatsoever to the user, don't get why something this this unbalanced would be allowed to function in PVP, it made a 22 damage knife do 1.8 brute and a 12mm revolver (with damage mods) do ~2 brute. basically nullifies FBPs/Synths from combat with a person and prevents them from even using it cause "teehee"

**Comment by ChefDoggo** [comment]
funny game reference shouldn't outweigh actual gameplay balance when making an item/feature imo, otherwise i should be allowed to add some stupid terraria pistol that does like 400 brute per shot cause "well its a reference to a gun that does a shitton of damage!"

**Comment by Rat-Time** [comment]
If this does go through it should have the spawn rate of a horrible tie or a jeweled necklace from xenoarch.  As it is, it almost completely removes the threat from all robot/drone mobs, turning some of the challenge content we have into a cakewalk.  The pvp ramifications are even more drastic, anyone with core implants, implants, prosthetics, or FBPs in general would be dead in the water against someone wearing this.  This is said as someone who plays a character who could make full use of it but I wouldn't simply because it would make the game unfun.
That out of the way, it looks amazing and the code in it taught me some stuff.

**Comment by cdb-is-not-good** [comment]
Generally speaking we allow bluecrosses(and i guess crimson crosses by the same note) to be a bit OP. They're rare, they're random and in the case of crimson crosses often have downsides. I'm not overly concerned about the balance implications due to the existence of psions meaning just one of them has to be around to counter, but again- it's a top tier spawn, certainly it may need its spawn rate lowered though.

File: code/game/objects/random/oddities.dm [hunk 11]
```diff
@@ -87,6 +87,8 @@
 				//obj/item/material/butterfly/frenchman = 1,
 				//Gun/Tool Mods
 				/obj/item/gun_upgrade/mechanism/brass_kit = 1,
+				//Armor
+				/obj/item/clothing/suit/crimsoncross_regaloutfit = 1,
```
Comment (Trilbyspaceclone) [inline code/game/objects/random/oddities.dm:91]:
```suggestion
				/obj/item/clothing/suit/crimsoncross_regaloutfit = 0.1, //Little rarer then even soap
```

**Pull request merged**